### PR TITLE
JS: add support for unnamed/default exports in PackageExports.qll

### PR DESCRIPTION
**Pull request by erik-krogh**
Gets a TP/TN for CVE-2017-16117 

The support already existed for NodeJS modules. This adds the same thing for AMD/Closure modules.

### CODE REVIEW

**Comment by erik-krogh** [comment]
[An evaluation](https://git.semmle.com/erik/dist-compare-reports/tree/eriks-profile-machine_1611129750221) is surprisingly good.   
I think it hits a better optimization somehow.

File: javascript/ql/src/semmle/javascript/PackageExports.qll [hunk 1]
```diff
@@ -78,5 +78,12 @@ private DataFlow::Node getAValueExportedByPackage() {
 private DataFlow::Node getAnExportFromModule(Module mod) {
   result.analyze().getAValue() = mod.(NodeModule).getAModuleExportsValue()
   or
+  exists(Variable var | var = mod.(Closure::ClosureModule).getExportsVariable() |
```
Comment (esbena) [inline javascript/ql/src/semmle/javascript/PackageExports.qll:81]: Do we have a test for this Closure case?

Comment (erik-krogh) [reply javascript/ql/src/semmle/javascript/PackageExports.qll:81]: Nope. I'll get on it.

**Comment by esbena** [comment]
LGTM. Could you perhaps check on a single project if the surprise-optimization still applies?

**Review by esbena — APPROVED** [review]

**Comment by erik-krogh** [comment]
> LGTM. Could you perhaps check on a single project if the surprise-optimization still applies?

Seems like the surprise-optimization was just a fluke. 
[A new evaluation just shows neutral performance](https://git.semmle.com/erik/dist-compare-reports/tree/erik-laptop_1611211157521).   

I also tried [a smaller evaluation](https://git.semmle.com/erik/dist-compare-reports/tree/erik-laptop_1611218140975) with the previous commit, and I got nothing.    
Both of those were done on my laptop, so there should be no flukes.

**Pull request merged**